src/client/play/js/events.js

```diff
@@ -12,6 +12,21 @@ import Tile from './tile';
 import tiles from './tiles';
 import ui from './ui';
 
+const KEY_UP = "KeyW";
+const KEY_DOWN = "KeyS";
+const KEY_LEFT = "KeyA";
+const KEY_RIGHT = "KeyD";
```

Comment:
Doesn't this make it so that AZERTY layout won't work anymore?

WASD on a QWERTY keyboard translates to ZQSD on an AZERTY keyboard 🤔

Reply:
> Doesn't this make it so that AZERTY layout won't work anymore?
> 
> WASD on a QWERTY keyboard translates to ZQSD on an AZERTY keyboard 🤔

No, both keyboard layouts will share the same `code` value.

Reply:
Quoting from [this article](https://hacks.mozilla.org/2017/03/internationalize-your-keyboard-controls/):
> So let’s say the user presses the Q key on a QWERTY keyboard. Then event.code gives you 'KeyQ' while event.key gives you 'q'.
> 
> But when a AZERTY keyboard user presses the A key, he also gets 'KeyQ' as event.code, yet event.key contains 'a'. This happens because the A key on a AZERTY keyboard is at the same location as the Q key on a QWERTY keyboard.

Reply:
Well then that is exactly what we want. Nice find! 🙂